Python/bltinmodule.c

```diff
@@ -46,14 +46,86 @@ _Py_IDENTIFIER(stderr);
 
 #include "clinic/bltinmodule.c.h"
 
+static PyObject*
+update_bases(PyObject* bases, PyObject** args, int nargs, int* modified_bases)
+{
+    int i, j, ind, tot_extra = 0;
+    PyObject *base, *meth, *new_base, *new_sub_base, *new_bases;
+    PyObject *stack[1] = {bases};
+    assert(PyTuple_Check(bases));
+
+    /* We have a separate cycle to calculate replacements with the idea that in
+       most cases we just scroll quickly though it and return original bases */
+    for (i = 2; i < nargs; i++){
```

Comment:
Why from 2?

Reply:
First two items in ``args`` are metaclass and name IIRC.